### PR TITLE
Add Doc Authors option

### PR DESCRIPTION
**Pull request by csestito**
## Description

- Added `authors` to frontmatter. In desktop it displays in the TOC, in mobile it displays at the top of the doc. 

- Authors takes a list of Github IDs. By default it uses the github ID as the name and github.com/github_id as the URL. `authors.json` file can be used to overwrite this with fields for `name`, `title`, `image_url`, and `url`. If image 404's it defaults to Dev at static/img/dev.png. 

- Added footer to encourage contributing

- Added authors to some files, will encourage others to add themselves to their docs. Will also check with authors for updates on their url, title, and image before pushing. 

- Updated contributing doc to add info on authors frontmatter 

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/theme/AuthorsSide/index.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import clsx from "clsx";
+import { useDoc } from "@docusaurus/theme-common/internal";
+import Link from "@docusaurus/Link";
+// Component responsible for the authors layout
+import globalAuthors from "../../../authors.json";
```
Comment (sserrata) [inline src/theme/AuthorsSide/index.js:6]: Could possibly switch this to `@site`

File: src/theme/AuthorsSide/index.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+              <MaybeLink href={author.url} className="avatar__photo-link">
+                <img
+                  className="avatar__photo"
+                  src={author.image_url}
+                  alt={author.name}
+                  onError={handleOnError}
+                />
+              </MaybeLink>
+            )}
+            {author.name && (
```
Comment (sserrata) [inline src/theme/AuthorsSide/index.js:72]: In line `35` you default to an empty string which could eval to `true` here - not sure if that's intended

File: src/theme/AuthorsSide/index.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                itemProp="author"
+                itemScope
+                itemType="https://schema.org/Person"
+              >
+                <div className="avatar__name">
+                  <MaybeLink href={author.url} itemProp="url">
+                    <span itemProp="name">{author.name}</span>
+                  </MaybeLink>
+                </div>
+                {author.title && (
```
Comment (sserrata) [inline src/theme/AuthorsSide/index.js:84]: In line `36` you fall back to empty string

File: firebase.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
...
+            "value": "default-src 'self'; script-src 'report-sample' 'self' 'sha256-HEXSlCvj5t1knUX5S9reED7mj347MrX5NNWmhVKV3AY=' 'sha256-LJv39KYSfXELQ23XLwGsxKqh55fWlLAveXNhE4GJztE=' 'sha256-zkIfJey2QJSMWsoE/xLvQ6GebR1o8N9s0f9cjTQ7mS0=' https://www.google-analytics.com/analytics.js https://www.googletagmanager.com/gtag/js https://www.googletagmanager.com/debug https://www.google.com/recaptcha/enterprise.js https://www.gstatic.com; style-src 'report-sample' 'self' 'unsafe-inline' https://fonts.googleapis.com https://use.fontawesome.com pan.dev; object-src 'none'; base-uri 'self'; font-src 'self' data: https://fonts.gstatic.com https://use.fontawesome.com; img-src 'self' 'unsafe-allow-redirects' data: https://cdn-images-1.medium.com https://medium.com https://www.datocms-assets.com https://www.google-analytics.com https://pan.dev https://raw.githubusercontent.com https://googletagmanager.com https://avatars.githubusercontent.com; worker-src 'none'; connect-src https://www.google-analytics.com https://*.paloaltonetworks.com https://cors.pan.dev https://*.algolia.net https://*.googleapis.com; frame-src https://www.google.com;"
```
Comment (sserrata) [inline firebase.json:24]: Cleanup `unsafe-allow-redirects` if no longer needed/required

File: src/css/custom.scss [hunk 5]
```diff
@@ -709,3 +709,7 @@ a {
   transform: rotate(180deg) !important;
 }
 /* END DETAILS MARKERS */
+.avatar__photo {
```
Comment (sserrata) [inline src/css/custom.scss:712]: Consider using a custom class to set these styles so the changes are only applied to the `author__avatar`

**Comment by csestito** [comment]
Experimental - Added support to put Authors in footer of docs without a TOC. 

See https://pan-dev-f1b58--pr313-lugpd57b.web.app/ansible/docs/panos/info/connectivity/ for an example.

**Pull request closed without merging**